lighthouse-viewer/app/src/lighthouse-report-viewer.js

```diff
@@ -284,6 +301,44 @@ class LighthouseViewerReport {
   }
 
   /**
+   * Handles changes to the gist url input.
+   */
+  onInputChange(e) {
+    e.stopPropagation();
+
+    if (!e.target.value) {
+      return;
+    }
+
+    try {
+      this._loadFromGistURL(e.target.value);
+    } catch (err) {
+      logger.error('Invalid URL');
+    }
+  }
+
+  /**
+   * Updates URL with user's gist and loads from github.
+   * @param {string} url Gist URL.
+   */
+  _loadFromGistURL(url) {
+    try {
+      url = new URL(url);
+
+      if (url.origin !== 'https://gist.github.com') {
+        logger.error('URL was not a gist');
+        return;
+      }
+
+      const id = url.href.substring(url.href.lastIndexOf('/') + 1);
```

Comment:
want to pull it out with a regex instead? seems slightly more robust. 

i think the following would work..:

```js
new URL(location.href).pathname.match(/[a-f0-9]{5,}/)
```